### ISSUE TITLE
[Telemetry] Add application_channel to baseline ping

### ISSUE THREAD

**Issue by bbinto** [mozilla-mobile/fenix#927]
#### What questions will you answer with this data?
- As a product manager, I want to know all event and baseline metrics by channel so I can understand our Fenix user population by channel better

#### Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
- Separate our users by channels (nightly, beta, release)

#### What probes (suggested, if applicable)
- application channel, see row 31 in https://docs.google.com/spreadsheets/d/1P2_aGEE9OtjM3x71d0GBhnHDhzxEL02xEHuKn1no23E/edit#gid=0
- similar to Focus

### Dependencies (Added by PM and Eng)
- Core ping / baseline ping

### Acceptance Criteria (Added by PM)
- Baseline pings can be queried via re:dash
- We are sending telemetry events for the actions listed in the requirements
- We have documented the telemetry
- We have asked a data steward to [review](https://github.com/mozilla/data-review/blob/master/request.md) the telemetry
- NOT an AC: Data science to create dashboard or further graphs (this will be a separate issue, this issue is only about hooking up the events described and that they can be queried in re-dash)

**Comment by bbinto** [comment]
@vesta0 this can wait until after Beta, and only needs to be addressed before we go live with MVP production - however I think we agreed that's not needed anymore after our conversation last week.

**Comment by sarentz** [comment]
@Dexterp37 Is this the right thing to do if we want to be able to separate telemetry for our different *applications* ? We will have a separate Nightly, Beta and Release app. How should that work in terms of telemetry pings?

I also filed https://github.com/mozilla-mobile/fenix/issues/1728 which may be similar.

**Comment by travis79** [comment]
The `client_info` section of glean pings contains an optional field `app_channel` which can be set through the `channel` property in the configuration object passed into the `Glean.initialize()` function.  As far as I can tell this should ready to use, but it's up to the integrating app to determine the proper channel to pass.

**Comment by sarentz** [comment]
@colintheshots We talked about the 'make the app aware of channels' bug today - this is a good usecase for that. TO set the `channel` property at runtime to the actual channel. What do you think?

**Comment by Dexterp37** [comment]
Sorry for the delay, was on holiday.

> @Dexterp37 Is this the right thing to do if we want to be able to separate telemetry for our different _applications_ ? We will have a separate Nightly, Beta and Release app. How should that work in terms of telemetry pings?

As Travis pointed out, Glean supports [channels](https://github.com/mozilla-mobile/android-components/tree/master/components/service/glean#initializing-glean) out of the box. However, the application is responsible for communicating the channel it is on to Glean.

In addition to this, please note that Glean pings are submitted to a different endpoint depending on the `applicationId`: as far as I remember, Fenix has a different application id for beta/nightly/release. So, while setting a `channel` at startup is a good thing and a good practice, you might be able to tell things apart on the pipeline based on the application id as well.

tl;dr - setting the channel, in this specific case, is encouraged but not strictly required

> Also, assuming that whatever needs to happen will apply to all pings. (baseline, event, ...)

Yes, this applies to all pings, even custom ones.

> I also filed #1728 which may be similar.

**Comment by sarentz** [comment]
Thanks @Dexterp37 this makes sense. Just to confirm, the app id is also ending up in the tables right? As in, we could write a query that compares some metric between beta and release, based on the app id in the `WHERE` clause?

**Comment by Dexterp37** [comment]
> 
> 
> Thanks @Dexterp37 this makes sense. Just to confirm, the app id is also ending up in the tables right? As in, we could write a query that compares some metric between beta and release, based on the app id in the `WHERE` clause?

Yes, it will be possible to join different tables, as @fbertsch shows [in this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1536464#c4). I think it should help with your use-case.

**Comment by vesta0** [comment]
@sarentz @boek  I think we can close this bug since we are no longer following the separate release track model (Focus model) and instead will have different app store ids. If you agree, please go ahead and close this.

**Comment by sarentz** [comment]
I think we still need this because we will be shipping both Nightly and Release with the same app identifier for a little while until Nightly moves to it's own app?

**Comment by vesta0** [comment]
Thanks for clarifying @st3fan , yes sounds like we need to do it.  cc @boek

**Comment by sv-ohorvath** [comment]
Verified that "app_channel: release" appears in baseline, events, metrics and custom pings.
